src/recon.rs

```diff
@@ -1591,8 +1620,46 @@ enum CfSelect {
     Task,
 }
 
+impl CfSelect {
+    fn set<BD: BitDepth>(
+        self,
+        f: &Rav1dFrameData,
+        t: &mut Rav1dTaskContext,
```

Comment:
Now that I think about it, it's probably not a good idea to pass a `&mut Rav1dTaskContext` here just to use one field, i.e. you might want to take `t.cf` as the argument, rather than all of `t`. The current usages of this are dereferencing a `*mut Rav1dTaskContext` so you're not hitting borrow errors yet, but once `decode_coefs` is taking a `t: &mut Rav1dTaskContext` you're likely going to hit borrow errors trying to take a mutable reference to all of `t`.

Reply:
The same also probably applies to `get`, I think.

Reply:
Probably, but it's easy enough to fix IF we hit that, rather than making this a bit less clear preemptively IMO.

Reply:
I agree it'd be better to take `cf: {&,&mut} BitDepthUnion<Cf>` as an individual field here instead of all of `t`.  And I think it's clearer this way as well.